File: sdk/node/test_model_middleware_health_fixture.mjs
import assert from 'node:assert/strict'
import { mkdtemp, readFile, rm } from 'node:fs/promises'
import { tmpdir } from 'node:os'
import path from 'node:path'
import { fileURLToPath } from 'node:url'
import mod from './index.js'

const root = path.resolve(path.dirname(fileURLToPath(import.meta.url)), '..')
const fixture = JSON.parse(
  await readFile(path.join(root, 'evaluation', 'model-middleware-health-v1.json'), 'utf8'),
)

const INLINE_CONFIG = `
default_model = "openai/fixture-model"
providers "openai" {
  apiKey = "fixture-key-never-sent"
  baseUrl = "https://fixture.invalid/v1"
  models "fixture-model" {
    name = "Fixture Model"
  }
}
`.trim()

function assertNoForbiddenFields(value) {
  const forbidden = new Set(fixture.forbidden_fields)
  if (Array.isArray(value)) {
    for (const child of value) {
      assertNoForbiddenFields(child)
    }
    return
  }
  if (value && typeof value === 'object') {
    for (const [key, child] of Object.entries(value)) {
      assert.equal(forbidden.has(key), false, `forbidden diagnostic field ${key}`)
      assertNoForbiddenFields(child)
    }
  }
}

function assertSnapshot(health) {
  for (const field of fixture.required_snapshot_fields) {
    assert.ok(field in health, `missing snapshot field ${field}`)
    assert.equal(health[field], 0, `${field} must start at zero`)
  }
  assertNoForbiddenFields(health)
}

const workspace = await mkdtemp(path.join(tmpdir(), 'a3s-node-middleware-health-'))
try {
  const agent = await mod.Agent.create(INLINE_CONFIG)
  const session = await agent.session(workspace, {
    sessionId: 'node-middleware-health-fixture',
  })
  try {
    const health = session.modelMiddlewareHealth()
    assertSnapshot(health)
  } finally {
    await session.close()
    await agent.close()
  }
} finally {
  await rm(workspace, { recursive: true, force: true })
}

console.log('model middleware health fixture ok')

File: sdk/python/tests/test_model_middleware_health_fixture.py
"""Cross-SDK contract checks for secret-free middleware health."""

from __future__ import annotations

import asyncio
import json
import tempfile
from pathlib import Path
from typing import Any

from a3s_code import Agent, SessionOptions


FIXTURE = json.loads(
    (
        Path(__file__).resolve().parents[2]
        / "evaluation"
        / "model-middleware-health-v1.json"
    ).read_text(encoding="utf-8")
)
INLINE_CONFIG = """
default_model = "openai/fixture-model"
providers "openai" {
  apiKey = "fixture-key-never-sent"
  baseUrl = "https://fixture.invalid/v1"
  models "fixture-model" {
    name = "Fixture Model"
  }
}
""".strip()


def _assert_no_forbidden_fields(value: Any) -> None:
    forbidden = set(FIXTURE["forbidden_fields"])
    if isinstance(value, dict):
        for key, child in value.items():
            assert key not in forbidden, f"forbidden diagnostic field {key}"
            _assert_no_forbidden_fields(child)
    elif isinstance(value, list):
        for child in value:
            _assert_no_forbidden_fields(child)


def _assert_snapshot(health: dict[str, Any]) -> None:
    for field in FIXTURE["required_snapshot_fields"]:
        assert field in health, f"missing snapshot field {field}"
        assert health[field] == 0, f"{field} must start at zero"
    _assert_no_forbidden_fields(health)


def test_model_middleware_health_fixture() -> None:
    async def scenario() -> None:
        agent = await Agent.create_async(INLINE_CONFIG)
        with tempfile.TemporaryDirectory(prefix="a3s-python-middleware-health-") as workspace:
            options = SessionOptions()
            options.session_id = "python-middleware-health-fixture"
            session = await agent.session_async(workspace, options)
            try:
                health = session.model_middleware_health()
                assert isinstance(health, dict)
                _assert_snapshot(health)
            finally:
                await session.close_async()
                await agent.close_async()

    asyncio.run(scenario())
